genesyscloud/resource_genesyscloud_recording_media_retention_policy.go

```diff
@@ -0,0 +1,2715 @@
+		Schema: map[string]*schema.Schema{
+			"language_id": {
+				Description: "The language name.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+		},
+	}
+
+	timeSlot = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"start_time": {
+				Description: "start time in xx:xx:xx.xxx format",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"stop_time": {
+				Description: "stop time in xx:xx:xx.xxx format",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"day": {
+				Description: "Day for this time slot, Monday = 1 ... Sunday = 7",
+				Type:        schema.TypeInt,
+				Optional:    true,
+			},
+		},
+	}
+
+	timeAllowed = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"time_slots": {
+				Description: "",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        timeSlot,
+			},
+			"time_zone_id": {
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"empty": {
+				Description: "",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+		},
+	}
+
+	durationCondition = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"duration_target": {
+				Description:  "",
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"DURATION", "DURATION_RANGE"}, false),
+			},
+			"duration_operator": {
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"duration_range": {
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"duration_mode": {
+				Description: "",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+		},
+	}
+
+	callMediaPolicyConditions = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"for_users": {
+				Description: "",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        user,
+			},
+			"date_ranges": {
+				Description: "",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Default:     nil,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"for_queues": {
+				Description: "",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        queue,
+			},
+			"wrapup_codes": {
+				Description: "",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        wrapupCode,
+			},
+			"languages": {
+				Description: "",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        language,
+			},
+			"time_allowed": {
+				Description: "",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Elem:        timeAllowed,
+			},
+			"directions": {
+				Description: "",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Default:     nil,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				// Valid values: "INBOUND", "OUTBOUND"
```

Comment:
Anywhere that you have these `// Valid values` you can use the validate.StringInSlice to validate this